### PR TITLE
Catch exceptions when trying to load cpp plugins

### LINKED ISSUE #52146: 3.31.0-Master: Crash during plugin reinstall
### What is the bug or the crash?

Hi,
  I wanted to re-intall all plugins already installed as I swapped python versions (now 3.10). There is no possibility to select all currently installed plugins and reinstall them all at once. Anyway, I slowly went one after another and selected "Reinstall plugin", then with some it crashed. The info message about a plugin successfully reinstalled should contain name of the plugin, btw.

```
src/core/qgsmessagelog.cpp:29 : (logMessage) [6226ms] 2023-03-08T02:52:46 Messages[0] <b>Plugin reinstalled successfully</b>
QGIS died on signal 11src/app/qgscrashhandler.cpp:71 : (handleCrash) [9235ms] /tmp/qgis-crash-info-17889
src/app/qgscrashhandler.cpp:124 : (handleCrash) [0ms] /usr/bin/qgiscrashhandler
[New LWP 17890]
[New LWP 17895]
[New LWP 17896]
[New LWP 17897]
[New LWP 17898]
[New LWP 17899]
[New LWP 17900]
[New LWP 17902]
[New LWP 17905]
[New LWP 17908]
[New LWP 17909]
[New LWP 17910]
[New LWP 17911]
[New LWP 17997]
[New LWP 17998]
[New LWP 17999]
[New LWP 18000]
[Thread debugging using libthread_db enabled]
Using host libthread_db library "/lib64/libthread_db.so.1".
0x00007f47518f2e47 in wait4 () from /lib64/libc.so.6
[Current thread is 1 (Thread 0x7f474a088900 (LWP 17889))]
#0  0x00007f47518f2e47 in wait4 () from /lib64/libc.so.6
No symbol table info available.
#1  0x0000564fad7dcd76 in qgisCrash(int) ()
No symbol table info available.
#2  <signal handler called>
No symbol table info available.
#3  0x00007f47518aa50c in __pthread_kill_implementation () from /lib64/libc.so.6
No symbol table info available.
#4  0x00007f475185bed2 in raise () from /lib64/libc.so.6
No symbol table info available.
#5  <signal handler called>
No symbol table info available.
#6  0x00007f475515d178 in QAction::text() const () from /usr/lib64/libQt5Widgets.so.5
No symbol table info available.
#7  0x00007f4756759c81 in QgsShortcutsManager::actionByName(QString const&) const () from /usr/lib64/libqgis_gui.so.3.31.0
No symbol table info available.
#8  0x00007f475675a98d in QgsShortcutsManager::registerAction(QAction*, QString const&) () from /usr/lib64/libqgis_gui.so.3.31.0
No symbol table info available.
#9  0x00007f46f6770975 in meth_QgisInterface_registerMainWindowAction () from /usr/lib/python3.10/site-packages/qgis/_gui.so
No symbol table info available.
#10 0x00007f47144f4b73 in cfunction_call () from /usr/lib64/libpython3.10.so.1.0
No symbol table info available.
#11 0x00007f47144b78d0 in _PyObject_MakeTpCall () from /usr/lib64/libpython3.10.so.1.0
No symbol table info available.
#12 0x00007f471446e074 in _PyEval_EvalFrameDefault () from /usr/lib64/libpython3.10.so.1.0
No symbol table info available.
#13 0x00007f4714589e74 in _PyEval_Vector () from /usr/lib64/libpython3.10.so.1.0
No symbol table info available.
#14 0x00007f471446b9a0 in _PyEval_EvalFrameDefault () from /usr/lib64/libpython3.10.so.1.0
No symbol table info available.
#15 0x00007f4714589e74 in _PyEval_Vector () from /usr/lib64/libpython3.10.so.1.0
No symbol table info available.
#16 0x00007f471446ae1c in _PyEval_EvalFrameDefault () from /usr/lib64/libpython3.10.so.1.0
No symbol table info available.
#17 0x00007f4714589e74 in _PyEval_Vector () from /usr/lib64/libpython3.10.so.1.0
No symbol table info available.
#18 0x00007f471446ae1c in _PyEval_EvalFrameDefault () from /usr/lib64/libpython3.10.so.1.0
No symbol table info available.
#19 0x00007f4714589e74 in _PyEval_Vector () from /usr/lib64/libpython3.10.so.1.0
No symbol table info available.
#20 0x00007f47144ba3d8 in method_vectorcall () from /usr/lib64/libpython3.10.so.1.0
No symbol table info available.
#21 0x00007f471446bfce in _PyEval_EvalFrameDefault () from /usr/lib64/libpython3.10.so.1.0
No symbol table info available.
#22 0x00007f4714589e74 in _PyEval_Vector () from /usr/lib64/libpython3.10.so.1.0
No symbol table info available.
#23 0x00007f471458a031 in PyEval_EvalCode () from /usr/lib64/libpython3.10.so.1.0
No symbol table info available.
#24 0x00007f47145c91a3 in run_eval_code_obj () from /usr/lib64/libpython3.10.so.1.0
No symbol table info available.
#25 0x00007f47145c93c6 in run_mod () from /usr/lib64/libpython3.10.so.1.0
No symbol table info available.
#26 0x00007f47145cbb75 in PyRun_StringFlags () from /usr/lib64/libpython3.10.so.1.0
No symbol table info available.
#27 0x00007f4723f11beb in QgsPythonUtilsImpl::runStringUnsafe(QString const&, bool) () from /usr/lib64/libqgispython.so.3.31.0
No symbol table info available.
#28 0x00007f4723f157d6 in QgsPythonUtilsImpl::runString(QString const&, QString, bool) () from /usr/lib64/libqgispython.so.3.31.0
No symbol table info available.
#29 0x00007f4757167bad in QgsPythonRunnerImpl::runCommand(QString, QString) () from /usr/lib64/libqgis_app.so.3.31.0
No symbol table info available.
#30 0x00007f475282089f in QgsPythonRunner::run(QString const&, QString const&) () from /usr/lib64/libqgis_core.so.3.31.0
No symbol table info available.
#31 0x00007f4757574cb1 in QgsPluginManager::buttonInstall_clicked() () from /usr/lib64/libqgis_app.so.3.31.0
No symbol table info available.
#32 0x00007f47546b1a2c in void doActivate<false>(QObject*, int, void**) () from /usr/lib64/libQt5Core.so.5
No symbol table info available.
#33 0x00007f4755256ac2 in QAbstractButton::clicked(bool) () from /usr/lib64/libQt5Widgets.so.5
No symbol table info available.
#34 0x00007f4755256d3a in QAbstractButtonPrivate::emitClicked() () from /usr/lib64/libQt5Widgets.so.5
No symbol table info available.
#35 0x00007f47552588b9 in QAbstractButtonPrivate::click() () from /usr/lib64/libQt5Widgets.so.5
No symbol table info available.
#36 0x00007f4755258adb in QAbstractButton::mouseReleaseEvent(QMouseEvent*) () from /usr/lib64/libQt5Widgets.so.5
No symbol table info available.
#37 0x00007f47551a52d8 in QWidget::event(QEvent*) () from /usr/lib64/libQt5Widgets.so.5
No symbol table info available.
#38 0x00007f4755162fbe in QApplicationPrivate::notify_helper(QObject*, QEvent*) () from /usr/lib64/libQt5Widgets.so.5
No symbol table info available.
#39 0x00007f475516b635 in QApplication::notify(QObject*, QEvent*) () from /usr/lib64/libQt5Widgets.so.5
No symbol table info available.
#40 0x00007f475263415b in QgsApplication::notify(QObject*, QEvent*) () from /usr/lib64/libqgis_core.so.3.31.0
No symbol table info available.
#41 0x00007f475467e748 in QCoreApplication::notifyInternal2(QObject*, QEvent*) () from /usr/lib64/libQt5Core.so.5
No symbol table info available.
#42 0x00007f4755169585 in QApplicationPrivate::sendMouseEvent(QWidget*, QMouseEvent*, QWidget*, QWidget*, QWidget**, QPointer<QWidget>&, bool, bool) () from /usr/lib64/libQt5Widgets.so.5
No symbol table info available.
#43 0x00007f47551be0b0 in QWidgetWindow::handleMouseEvent(QMouseEvent*) () from /usr/lib64/libQt5Widgets.so.5
No symbol table info available.
#44 0x00007f47551c13d0 in QWidgetWindow::event(QEvent*) () from /usr/lib64/libQt5Widgets.so.5
No symbol table info available.
#45 0x00007f4755162fbe in QApplicationPrivate::notify_helper(QObject*, QEvent*) () from /usr/lib64/libQt5Widgets.so.5
No symbol table info available.
#46 0x00007f475263415b in QgsApplication::notify(QObject*, QEvent*) () from /usr/lib64/libqgis_core.so.3.31.0
No symbol table info available.
#47 0x00007f475467e748 in QCoreApplication::notifyInternal2(QObject*, QEvent*) () from /usr/lib64/libQt5Core.so.5
No symbol table info available.
#48 0x00007f4754b18ddf in QGuiApplicationPrivate::processMouseEvent(QWindowSystemInterfacePrivate::MouseEvent*) () from /usr/lib64/libQt5Gui.so.5
No symbol table info available.
#49 0x00007f4754af89eb in QWindowSystemInterface::sendWindowSystemEvents(QFlags<QEventLoop::ProcessEventsFlag>) () from /usr/lib64/libQt5Gui.so.5
No symbol table info available.
#50 0x00007f4743b2327a in xcbSourceDispatch(_GSource*, int (*)(void*), void*) () from /usr/lib64/libQt5XcbQpa.so.5
No symbol table info available.
#51 0x00007f474cb003fa in g_main_dispatch (context=0x564fadf03600) at ../glib-2.74.4/glib/gmain.c:3454
        dispatch = 0x7f4743b23260 <xcbSourceDispatch(_GSource*, int (*)(void*), void*)>
        prev_source = 0x564fae01fbc0
        begin_time_nsec = 0
        was_in_call = 2
        user_data = 0x0
        callback = 0x0
        cb_funcs = 0x0
        cb_data = 0x0
        need_destroy = 0
        source = 0x564fae01fbc0
        current = 0x564fae32d5d0
        i = 1
        __func__ = "g_main_dispatch"
#52 0x00007f474cb01422 in g_main_context_dispatch (context=0x564fadf03600) at ../glib-2.74.4/glib/gmain.c:4172
No locals.
#53 0x00007f474cb01624 in g_main_context_iterate (context=0x564fadf03600, block=1, dispatch=1, self=0x564fadf04800) at ../glib-2.74.4/glib/gmain.c:4248
        max_priority = 2147483647
        timeout = 28122
        some_ready = 1
        nfds = 6
        allocated_nfds = 6
        fds = 0x7f473c005730
        begin_time_nsec = 0
#54 0x00007f474cb016fc in g_main_context_iteration (context=0x564fadf03600, may_block=1) at ../glib-2.74.4/glib/gmain.c:4313
        retval = 0
#55 0x00007f47546d11f6 in QEventDispatcherGlib::processEvents(QFlags<QEventLoop::ProcessEventsFlag>) () from /usr/lib64/libQt5Core.so.5
No symbol table info available.
#56 0x00007f475467d22b in QEventLoop::exec(QFlags<QEventLoop::ProcessEventsFlag>) () from /usr/lib64/libQt5Core.so.5
No symbol table info available.
#57 0x00007f475536e36e in QDialog::exec() () from /usr/lib64/libQt5Widgets.so.5
No symbol table info available.
#58 0x00007f46f62c595f in meth_QgsPluginManagerInterface_showPluginManager () from /usr/lib/python3.10/site-packages/qgis/_gui.so
No symbol table info available.
#59 0x00007f47144f4b73 in cfunction_call () from /usr/lib64/libpython3.10.so.1.0
No symbol table info available.
#60 0x00007f47144b78d0 in _PyObject_MakeTpCall () from /usr/lib64/libpython3.10.so.1.0
No symbol table info available.
#61 0x00007f471446e074 in _PyEval_EvalFrameDefault () from /usr/lib64/libpython3.10.so.1.0
No symbol table info available.
#62 0x00007f4714589e74 in _PyEval_Vector () from /usr/lib64/libpython3.10.so.1.0
No symbol table info available.
#63 0x00007f47144ba3d8 in method_vectorcall () from /usr/lib64/libpython3.10.so.1.0
No symbol table info available.
#64 0x00007f471446e02e in _PyEval_EvalFrameDefault () from /usr/lib64/libpython3.10.so.1.0
No symbol table info available.
#65 0x00007f4714589e74 in _PyEval_Vector () from /usr/lib64/libpython3.10.so.1.0
No symbol table info available.
#66 0x00007f471458a031 in PyEval_EvalCode () from /usr/lib64/libpython3.10.so.1.0
No symbol table info available.
#67 0x00007f47145c91a3 in run_eval_code_obj () from /usr/lib64/libpython3.10.so.1.0
No symbol table info available.
#68 0x00007f47145c93c6 in run_mod () from /usr/lib64/libpython3.10.so.1.0
No symbol table info available.
#69 0x00007f47145cbb75 in PyRun_StringFlags () from /usr/lib64/libpython3.10.so.1.0
No symbol table info available.
#70 0x00007f4723f11beb in QgsPythonUtilsImpl::runStringUnsafe(QString const&, bool) () from /usr/lib64/libqgispython.so.3.31.0
No symbol table info available.
#71 0x00007f4723f157d6 in QgsPythonUtilsImpl::runString(QString const&, QString, bool) () from /usr/lib64/libqgispython.so.3.31.0
No symbol table info available.
#72 0x00007f4757167bad in QgsPythonRunnerImpl::runCommand(QString, QString) () from /usr/lib64/libqgis_app.so.3.31.0
No symbol table info available.
#73 0x00007f475282089f in QgsPythonRunner::run(QString const&, QString const&) () from /usr/lib64/libqgis_core.so.3.31.0
No symbol table info available.
#74 0x00007f47570d4bfd in QgisApp::showPluginManager() () from /usr/lib64/libqgis_app.so.3.31.0
No symbol table info available.
#75 0x00007f47546b1a2c in void doActivate<false>(QObject*, int, void**) () from /usr/lib64/libQt5Core.so.5
No symbol table info available.
#76 0x00007f475515c782 in QAction::triggered(bool) () from /usr/lib64/libQt5Widgets.so.5
No symbol table info available.
#77 0x00007f475515f3ab in QAction::activate(QAction::ActionEvent) () from /usr/lib64/libQt5Widgets.so.5
No symbol table info available.
#78 0x00007f47552e6572 in QMenuPrivate::activateCausedStack(QVector<QPointer<QWidget> > const&, QAction*, QAction::ActionEvent, bool) () from /usr/lib64/libQt5Widgets.so.5
No symbol table info available.
#79 0x00007f47552ee4e4 in QMenuPrivate::activateAction(QAction*, QAction::ActionEvent, bool) () from /usr/lib64/libQt5Widgets.so.5
No symbol table info available.
#80 0x00007f47551a52d8 in QWidget::event(QEvent*) () from /usr/lib64/libQt5Widgets.so.5
No symbol table info available.
#81 0x00007f4755162fbe in QApplicationPrivate::notify_helper(QObject*, QEvent*) () from /usr/lib64/libQt5Widgets.so.5
No symbol table info available.
#82 0x00007f475516b635 in QApplication::notify(QObject*, QEvent*) () from /usr/lib64/libQt5Widgets.so.5
No symbol table info available.
#83 0x00007f475263415b in QgsApplication::notify(QObject*, QEvent*) () from /usr/lib64/libqgis_core.so.3.31.0
No symbol table info available.
#84 0x00007f475467e748 in QCoreApplication::notifyInternal2(QObject*, QEvent*) () from /usr/lib64/libQt5Core.so.5
No symbol table info available.
#85 0x00007f4755169585 in QApplicationPrivate::sendMouseEvent(QWidget*, QMouseEvent*, QWidget*, QWidget*, QWidget**, QPointer<QWidget>&, bool, bool) () from /usr/lib64/libQt5Widgets.so.5
No symbol table info available.
#86 0x00007f47551be506 in QWidgetWindow::handleMouseEvent(QMouseEvent*) () from /usr/lib64/libQt5Widgets.so.5
No symbol table info available.
#87 0x00007f47551c13d0 in QWidgetWindow::event(QEvent*) () from /usr/lib64/libQt5Widgets.so.5
No symbol table info available.
#88 0x00007f4755162fbe in QApplicationPrivate::notify_helper(QObject*, QEvent*) () from /usr/lib64/libQt5Widgets.so.5
No symbol table info available.
#89 0x00007f475263415b in QgsApplication::notify(QObject*, QEvent*) () from /usr/lib64/libqgis_core.so.3.31.0
No symbol table info available.
#90 0x00007f475467e748 in QCoreApplication::notifyInternal2(QObject*, QEvent*) () from /usr/lib64/libQt5Core.so.5
No symbol table info available.
#91 0x00007f4754b18ddf in QGuiApplicationPrivate::processMouseEvent(QWindowSystemInterfacePrivate::MouseEvent*) () from /usr/lib64/libQt5Gui.so.5
No symbol table info available.
#92 0x00007f4754af89eb in QWindowSystemInterface::sendWindowSystemEvents(QFlags<QEventLoop::ProcessEventsFlag>) () from /usr/lib64/libQt5Gui.so.5
No symbol table info available.
#93 0x00007f4743b2327a in xcbSourceDispatch(_GSource*, int (*)(void*), void*) () from /usr/lib64/libQt5XcbQpa.so.5
No symbol table info available.
#94 0x00007f474cb003fa in g_main_dispatch (context=0x564fadf03600) at ../glib-2.74.4/glib/gmain.c:3454
        dispatch = 0x7f4743b23260 <xcbSourceDispatch(_GSource*, int (*)(void*), void*)>
        prev_source = 0x0
        begin_time_nsec = 0
        was_in_call = 0
        user_data = 0x0
        callback = 0x0
        cb_funcs = 0x0
        cb_data = 0x0
        need_destroy = 0
        source = 0x564fae01fbc0
        current = 0x564fae32d5d0
        i = 1
        __func__ = "g_main_dispatch"
#95 0x00007f474cb01422 in g_main_context_dispatch (context=0x564fadf03600) at ../glib-2.74.4/glib/gmain.c:4172
No locals.
#96 0x00007f474cb01624 in g_main_context_iterate (context=0x564fadf03600, block=1, dispatch=1, self=0x564fadf04800) at ../glib-2.74.4/glib/gmain.c:4248
        max_priority = 2147483647
        timeout = 483
        some_ready = 1
        nfds = 6
        allocated_nfds = 6
        fds = 0x7f473c005730
        begin_time_nsec = 0
#97 0x00007f474cb016fc in g_main_context_iteration (context=0x564fadf03600, may_block=1) at ../glib-2.74.4/glib/gmain.c:4313
        retval = 0
#98 0x00007f47546d11f6 in QEventDispatcherGlib::processEvents(QFlags<QEventLoop::ProcessEventsFlag>) () from /usr/lib64/libQt5Core.so.5
No symbol table info available.
#99 0x00007f475467d22b in QEventLoop::exec(QFlags<QEventLoop::ProcessEventsFlag>) () from /usr/lib64/libQt5Core.so.5
No symbol table info available.
#100 0x00007f47546853b0 in QCoreApplication::exec() () from /usr/lib64/libQt5Core.so.5
No symbol table info available.
#101 0x0000564fad7d93df in main ()
No symbol table info available.
[Inferior 1 (process 17889) detached]
gdb returned 0
Neúspěšně ukončen (SIGABRT) (core dumped [obraz paměti uložen])
```

### Steps to reproduce the issue

Update a plugin, cannot say which it was was.

### Versions

<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.0//EN" "http://www.w3.org/TR/REC-html40/strict.dtd">
<html><head><meta http-equiv="Content-Type" content="text/html; charset=utf-8" /><style type="text/css">
p, li { white-space: pre-wrap; }
</style></head><body>

QGIS version | 3.31.0-Master | QGIS code revision | 771c85ac2a
-- | -- | -- | --
Qt version | 5.15.8
Python version | 3.10.10
GDAL/OGR version | 3.6.2
PROJ version | 9.1.1
EPSG Registry database version | v10.076 (2022-08-31)
GEOS version | 3.11.1-CAPI-1.17.1
SQLite version | 3.40.1
PDAL version | 2.5.0
PostgreSQL client version | unknown
SpatiaLite version | 5.0.1
QWT version | 6.1.5
QScintilla2 version | 2.13.3
OS version | Gentoo Linux
  |   |   |  
This copy of QGIS writes debugging output.
  |   |   |  
Active Python plugins
poiexport | 3.0.4
CalculateGeometry | 0.6.8
latlontools | 3.6.8
polygonclickmap | 0.7
raster_divider | 1.1
copy_coords | 0.2
clipper | 1.2
vector_correction | 0.0.2
qProf | 0.5.0
PointConnector | 2.0
lftools | 1.10.0
kmltools | 3.1.27
AnotherDXF2Shape | 1.2.5
Serval | 3.10.5
GpxSegmentImporter | 2.5
raster_tracer | 0.3.2
czech_slovak_freegeodata | 0.6
localcoordinates | 0.3
shapefile_encoding_fixer | 0.7
geometric_attributes | 0.3
VectorMCDA | 3.0.1
redLayer | 2.2
opeNoise | 2.0
ThresholdVect | 0.1
layer2kmz | 1.5.3
rasterdataplotting | 1.6.3
qgeric | 1.91
FreehandRasterGeoreferencer | 0.8.3
qgis-ruian-plugin | 1.1.3
raster_cutter | 0.6
Coordtransform | 1.1
shapetools | 3.4.10
ImportPhotos | 3.0.5
line_profile | 3.1.2
shp_to_obs | 1.1.3
freehandEditing3 | version 1.1.0
processing | 2.12.99
db_manager | 0.1.20
grassprovider | 2.12.99
MetaSearch | 0.3.6

</body></html>

### Supported QGIS version

- [ ] I'm running a supported QGIS version according to the roadmap.

### New profile

- [ ] I tried with a new QGIS profile

### Additional context

Please improve the error messages at least.

### PR DESCRIPTION
**Pull request by nyalldawson**
Fixes crash when showing the plugin manager when a cpp plugin raises an unhandled exception

Fixes https://github.com/qgis/QGIS/issues/52146

### CODE REVIEW

**Comment by nyalldawson** [comment]
@3nids this crash is revealing a deeper problem -- specifically the exception which is not being caught is a QgsSettingsException "Settings tree node 'babel-devices' already holds a child setting with key 'wptdownload'. 

I encounter this one when the libqgisgrass8 plugin is trying to be loaded.

File: src/app/pluginmanager/qgspluginmanager.cpp [hunk 1]
```diff
@@ -405,121 +405,135 @@ void QgsPluginManager::getCppPluginsMetadata()
...
-      }
-      if ( pUpdateDate )
-      {
-        QgsDebugMsgLevel( "Plugin update date: " + *pUpdateDate(), 2 );
-      }
-      else
-      {
-        QgsDebugMsgLevel( QStringLiteral( "Plugin update date not returned when queried" ), 2 );
+        QgsDebugMsg( QStringLiteral( "Unhandled settings exception loading %1: %2" ).arg( lib, ex.what() ) );
+        continue;
```
Comment (nirvn) [inline src/app/pluginmanager/qgspluginmanager.cpp:527]: @nyalldawson , is there a need to `delete myLib;` here?

Comment (nyalldawson) [reply src/app/pluginmanager/qgspluginmanager.cpp:522]: I've moved it to a unique_ptr now

**Comment by 3nids** [comment]
I fail a bit to understand what's happening here.
The setting is defined only once and is a static variable. 
Do you have an explanation?

**Pull request merged**